applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.cpp

```diff
@@ -578,31 +585,33 @@ namespace MPMParticleGeneratorUtility
                             // Check Normal direction
                             if (flip_normal_direction) mpc_normal *= -1.0;
 
+                            ProcessInfo process_info = ProcessInfo();
+
                             // Setting particle condition's initial condition
                             // TODO: If any variable is added or remove here, please add and remove also at the second loop below
-                            p_condition->SetValue(MPC_CONDITION_ID, mpc_condition_id);
-                            p_condition->SetValue(MPC_COORD, mpc_xg);
-                            p_condition->SetValue(MPC_AREA, mpc_area);
-                            p_condition->SetValue(MPC_NORMAL, mpc_normal);
-                            p_condition->SetValue(MPC_DISPLACEMENT, mpc_displacement);
-                            p_condition->SetValue(MPC_IMPOSED_DISPLACEMENT, mpc_imposed_displacement);
-                            p_condition->SetValue(MPC_VELOCITY, mpc_velocity);
-                            p_condition->SetValue(MPC_IMPOSED_VELOCITY, mpc_imposed_velocity);
-                            p_condition->SetValue(MPC_ACCELERATION, mpc_acceleration);
-                            p_condition->SetValue(MPC_IMPOSED_ACCELERATION, mpc_imposed_acceleration);
+                            //p_condition->SetValueOnIntegrationPoints(MPC_CONDITION_ID, mpc_condition_id, process_info);
```

Comment:
hmmm, why we dont need the id?

Reply:
Through the CalculateOnIntegrationPoints we can directly access it. Thus, no need to duplicate it.

Reply:
@tteschemacher Can you check the coupling test too implemented in CoSimulation Application?